ui/js/controllers/logviewer.js

```diff
@@ -33,13 +32,58 @@ logViewerApp.controller('LogviewerCtrl', [
         $scope.currentLineNumber = 0;
         $scope.highestLine = 0;
         $scope.showSuccessful = true;
-
+        $scope.willScroll = false;
         $scope.$watch('artifact', function () {
             if (!$scope.artifact) {
                 return;
             }
             $scope.showSuccessful = !$scope.hasFailedSteps();
         });
+        $scope.$watch('[selectedBegin, selectedEnd]', function(newVal, oldVal) {
+            var newHash = (newVal[0] == newVal[1])? newVal[0] : newVal[0] + "-L" + newVal[1];
+            if (!isNaN(newVal[0])) {
+                $location.hash("L" + newHash);
+            }
+            else {
+                if($scope.artifact) {
```

Comment:
Same here as per below, we'd like a space after the `if`
`if ($scope.artifact)`